index.js

```diff
@@ -335,10 +338,11 @@
   var TypeRep = $.NullaryType(
     'sanctuary/TypeRep',
     function(x) {
-      return R.type(x) === 'Function' ||
-             (x != null &&
-              R.type(x.name) === 'String' &&
-              R.type(x.test) === 'Function');
+      return R.type(x) === 'Function'
+          && R.type(global_[x.name]) === 'Function'
+          || x != null
+          && x.prototype != null
+          && R.type(x.prototype['@@type']) === 'String';
```

Comment:
Is it sort of breaking change? if one was using  `data.task` for example user should itself add the `@@type` prop to it's prototype